tests/cases/fourslash/completionInsideObjectLiteralExpressionWithInstantiatedClassType.ts

```diff
@@ -0,0 +1,38 @@
+﻿/// <reference path="fourslash.ts" />
+
+////class C1 {
+////    public a: string;
+////    protected b: string;
+////    private c: string;
+////
+////    constructor(a: string, b = "", c = "") {
+////        this.a = a;
+////        this.b = b;
+////        this.c = c;
+////    }
+////}
+////class C2 {
+////    public a: string;
+////    constructor(a: string) {
+////        this.a = a;
+////    }
+////}
+////function f1(foo: C1 | { d: number }) {}
+////f1({ /*1*/ });
+////
+////function f2(foo: C1) {}
+////f2({ /*2*/ });
+////
+////function f3(foo: C1 | C2) {}
+////f3({ /*3*/ });
+
+verify.completions({
+    marker: "1",
+    exact: ["d"]
+}, {
+    marker: "2",
+    exact: []
+}, {
+    marker: "3",
+    exact: []

```

Comment:
IOW this should be `"a"`

Reply:
@RyanCavanaugh Thanks for the feedback.

> No Dish property should be autocompleted. In my example, the autocompletion list should be empty when opening an object. 

I though that main concern to to avoid issues like so https://www.typescriptlang.org/play?#code/MYGwhgzhAEDCCM0DeBYAUNTAHArgIxAEthowAuaCAFwCdCA7AcwG51NosaB7KgU2D4ATaHgrU6TVhmx0AbmD7RgY2gxbo2mYF3ricArjQAU5SqqYAaEdAC80AET2rJO44CUyTe0xUAFoQgAOjBbUilvdj8AwLxQvHCI6CiglyUE6ABfdCy0dAAzHHoBQh1oPPgjPK4uCgQPJBzyoyRSCkdMtykgA, 

```ts
class C1 {
   public a: string;
   protected b: string;
   private c: string;

   constructor(a: string, b = "", c = "") {
       this.a = a;
       this.b = b;
       this.c = c;
   }
}

function f1(foo: C1) {}
f1({ a: "" }); 
// Argument of type '{ a: string; }' is not assignable to parameter of type 'C1'.
//  Type '{ a: string; }' is missing the following properties from type 'C1': b, c(2345)
```

Reply:
For `C1`, as shown here, yes.

For this example, though, we should still show `'a'`
```ts
class C4 {
   public a: string;
}

function f4(foo: C4) {}
f4({ /* completion here*/ }); 
```

Reply:
Oke, thanks!